private/voucher_private.h

```diff
@@ -400,7 +400,9 @@ dispatch_queue_create_with_accounting_override_voucher(const char *label,
  * voucher ports directly.
  */
 
+#ifdef HAVE_MACH
```

Comment:
we cannot have these types of checks inAPI headers, they will not pass on OSX (nobody defines HAVE_MACH). I would use either  `__APPLE__` or the `TARGET_OS_MAC` macro from TargetConditionals.h (there is also a `TARGET_OS_UNIX` there that might make sense for the 'linux-specific' checks)